buildconfig/CMake/CommonSetup.cmake

```diff
@@ -80,56 +80,65 @@ add_definitions(-DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG)
 # Silence issues with deprecated allocator methods in boost regex
 add_definitions(-D_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING)
 
+if(MANTID_FRAMEWORK_LIB STREQUAL "BUILD" OR MANTID_QT_LIB STREQUAL "BUILD")
 find_package(Poco 1.4.6 REQUIRED)
 add_definitions(-DPOCO_ENABLE_CPP11)
 
-find_package(GSL REQUIRED)
-find_package(Nexus 4.3.1 REQUIRED)
-find_package(MuParser REQUIRED)
-find_package(JsonCPP 0.7.0 REQUIRED)
+find_package(TBB REQUIRED)
+
+find_package(OpenSSL REQUIRED)
+
+endif()
+# if we are building the framework we will need these libraries.
+if (MANTID_FRAMEWORK_LIB STREQUAL "BUILD")
```

Comment:
Ditto about splitting this into it's own Framework_Lib cmake file (if it's trivial)